packages/flutter_tools/lib/src/commands/symbolize.dart

```diff
@@ -63,16 +72,50 @@ class SymbolizeCommand extends FlutterCommand {
   @override
   bool get shouldUpdateCache => false;
 
+  Map<int, String> _unitDebugInfoPathMap() {
+    final Map<int, String> map = <int, String>{};
+    final String? rootInfo = stringArg('debug-info');
+    if (rootInfo != null) {
+      map[rootLoadingUnitId] = rootInfo;
+    }
+    for (final String arg in stringsArg('unit-id-debug-info')) {
+      final int separatorIndex = arg.indexOf(':');
+      if (separatorIndex == -1) {
+        throwToolExit('The argument to "--unit-id-debug-info" must contain a unit ID and path,'
+            ' separated by ":": "$arg".');
+      }
+      final String unitIdString= arg.substring(0, separatorIndex);
+      final int? unitId = int.tryParse(unitIdString);
+      if (unitId == null) {
+        throwToolExit('The argument to "--unit-id-debug-info" must begin with'
+            ' a unit ID: "$unitIdString" is not an integer.');
+      }
+      final String unitDebugPath = arg.substring(separatorIndex + 1);
+      if (map.containsKey(unitId) && map[unitId] != unitDebugPath) {
+        throwToolExit('Different paths were given for the same loading unit'
+            ' $unitId: "${map[unitId]}" and "$unitDebugPath".');
+      }
+      map[unitId] = unitDebugPath;
```

Comment:
can we/should we validate this is a valid path? Not sure how to do that, but if it's just gonna fail later on, it might be helpful to fail here.

Reply:
Done, moved the checking on the map paths that was originally only in `validateCommand` and not in `runCommand` to this method instead. (Also have it returning a map of `File` references instead of `String`s now.